### PR TITLE
Maintain the set of linked SSTs in BlobFileMetaData

### PR DESCRIPTION
**Pull request by ltamasi**
Summary:
The `FileMetaData` objects associated with table files already contain the
number of the oldest blob file referenced by the SST in question. This patch
adds the inverse mapping to `BlobFileMetaData`, namely the set of table file
numbers for which the oldest blob file link points to the given blob file (these
are referred to as *linked SSTs*). This mapping will be used by the GC logic.

Implementation-wise, the patch builds on the `BlobFileMetaDataDelta`
functionality introduced in https://github.com/facebook/rocksdb/pull/6835: newly linked/unlinked SSTs are
accumulated in `BlobFileMetaDataDelta`, and the changes to the linked SST set
are applied in one shot when the new `Version` is saved. The patch also reworks
the blob file related consistency checks in `VersionBuilder` so they validate the
consistency of the forward table file -> blob file links and the backward blob file ->
table file links for blob files that are part of the `Version`.

Test Plan:
`make check`

### CODE REVIEW

File: db/version_builder.cc [hunk 1]
```diff
@@ -537,7 +615,23 @@ class VersionBuilder::Rep {
       }
     }
 
-    // Delete files
+    // Add new blob files
```
Comment (riversand963) [inline db/version_builder.cc:618]: The implementation implicitly requires addition/deletion of blob files to occur before addition of SSTs. Do you think  it's worth documenting?

**Comment by ltamasi** [comment]
Thanks for the review @riversand963 !

File: db/version_builder.cc [hunk 2]
```diff
@@ -430,6 +472,27 @@ class VersionBuilder::Rep {
     return base_vstorage_->GetFileLocation(file_number).GetLevel();
   }
 
+  uint64_t GetBlobFileNumberForTableFile(int level, uint64_t file_number) {
```
[the remark this answers is not in the document]
Comment (ltamasi) [reply db/version_builder.cc:475]: Sounds good. Will change.

File: db/version_builder.cc (hunk 1 above)
Comment (ltamasi) [reply db/version_builder.cc:618]: Good point, will add a comment to that effect.

**Pull request closed without merging**